tools/resourcedocsgen/pkg/docs/templates/function.tmpl

```diff
@@ -85,8 +85,8 @@ func </span>{{ index .FunctionName lGo }}Output<span class="p">(</span>{{ htmlSa
 <!-- Java -->
 <div>
 <pulumi-choosable type="language" values="java">
-<div class="highlight"><pre class="chroma"><code class="language-java" data-lang="java"><span class="k">public static CompletableFuture<{{ template "linkify_param" (index .FunctionResult lJava) }}> </span>{{ index .FunctionName lJava }}<span class="p">(</span>{{ htmlSafe (index .FunctionArgs lJava) }}<span class="p">)</span>
-<span class="k">public static Output<{{ template "linkify_param" (index .FunctionResult lJava) }}> </span>{{ index .FunctionName lJava }}<span class="p">(</span>{{ htmlSafe (index .FunctionArgs lJava) }}<span class="p">)</span>
+<div class="highlight"><pre class="chroma"><code class="language-java" data-lang="java"><span class="k">public static CompletableFuture<{{ template "linkify_param" (index .FunctionResult lJava) }}> </span>{{ index .FunctionName lJava }}<span class="p">(</span>{{ htmlSafe (index .FunctionArgs lJava) }}<span class="p">)</span>{{ if (index .HasOutputVersion lJava) }}
+<span class="k">public static Output<{{ template "linkify_param" (index .FunctionResult lJava) }}> </span>{{ index .FunctionName lJava }}<span class="p">(</span>{{ htmlSafe (index .FunctionArgsOutputVersion lJava) }}<span class="p">)</span>{{ end }}
```

Comment:
Do we believe that `.FunctionArgs` is always the same as `.FunctionArgsOutputVersion`, and that's why the docs won't change?

Can we construct a test case where they are different to put the `.FunctionArgs` -> `.FunctionArgsOutputVersion` change under test?

Reply:
Yes, both `FunctionArgs` and `FunctionArgsOutputVersion` are only written to in a single place by using the ret value of `mod.genFunctionArgs(f, funcNameMap, true/false /*outputVersion*/)`. 

The existing tests (e.g. `TestGeneratePackage`) already verify that we're generating the same output and do not regress. 

What would we gain here with a synthetic test that simulates a situation that cannot happen in practice?

Reply:
If we are sure that they are always the same, we can omit the test.